renderers/games/teams.py

```diff
@@ -78,11 +85,38 @@ def __render_team_text(canvas, layout, colors, team, homeaway, full_team_names,
     graphics.DrawText(canvas, font["font"], coords["x"], coords["y"], text_color_graphic, team_text)
 
 
-def __render_team_score(canvas, layout, colors, runs, homeaway, default_colors):
+def __render_score_component(canvas, layout, colors, homeaway, default_colors, coords, component_val, width):
+    # return actual calculated x coord.
     text_color = colors.get("text", default_colors["text"])
     text_color_graphic = graphics.Color(text_color["r"], text_color["g"], text_color["b"])
-    coords = layout.coords("teams.runs.{}".format(homeaway))
-    font = layout.font("teams.runs.{}".format(homeaway))
-    team_runs = str(runs)
-    team_runs_x = coords["x"] - (len(team_runs) * font["size"]["width"])
-    graphics.DrawText(canvas, font["font"], team_runs_x, coords["y"], text_color_graphic, team_runs)
+    font = layout.font(f"teams.runs.{homeaway}")  # just use the runs font
+    component_val = str(component_val)
+    std_spacing = True
+    for i in range(width):
+        score_x = coords["x"] - font["size"]["width"] + 1
+        coords["x"] = score_x
+        if len(component_val) > i:
+            c = component_val[len(component_val) - i - 1]
+            graphics.DrawText(canvas, font["font"], score_x, coords["y"], text_color_graphic, c)
+    return score_x - 1
+
+
+def __render_team_score(canvas, layout, colors, team, homeaway, default_colors, score_spacing, show_hits_errors):
+    if show_hits_errors:
+        coords = layout.coords(f"teams.runs.{homeaway}").copy()
+        x_calc = __render_score_component(
+            canvas, layout, colors, homeaway, default_colors, coords, team.errors, score_spacing["errors"]
+        )
+        coords["x"] = x_calc - 2
+        x_calc = __render_score_component(
+            canvas, layout, colors, homeaway, default_colors, coords, team.hits, score_spacing["hits"]
+        )
+        coords["x"] = x_calc - 2
+        x_calc = __render_score_component(
+            canvas, layout, colors, homeaway, default_colors, coords, team.runs, score_spacing["runs"]
+        )
+    else:
+        coords = layout.coords(f"teams.runs.{homeaway}").copy()
+        __render_score_component(
+            canvas, layout, colors, homeaway, default_colors, coords, team.runs, score_spacing["runs"]
+        )
```

Comment:
```suggestion
    __render_score_component(
        canvas, layout, colors, homeaway, default_colors, coords, team.runs, score_spacing["runs"]
    )
```

Drawing runs shouldn't need to be conditional, right? You can just do it following the if

Reply:
Yup, right again!